lib/iris/_lazy_data.py

```diff
@@ -158,3 +178,18 @@ def multidim_lazy_stack(stack):
         result = da.stack([multidim_lazy_stack(subarray)
                            for subarray in stack])
     return result
+
+
+def co_realise_cubes(cubes):
+    """
+    Fetch real data for multiple cubes at one time.
+
+    This fetches lazy content, equivalent to accessing each cube.data.
+    However, lazy calculations and data fetches can be shared between the
+    computations, improving performance.
+
+    """
+    results = _co_realise_lazy_arrays([cube.core_data() for cube in cubes])
```

Comment:
It seems like we should allow a combination of lazy and realised cubes to be passed to this method, and in fact I think the current code allows that (`da.compute()` accepts non-dask objects). Might it be worth mentioning this in the docstrings of this method and `_co_realise_lazy_arrays()` ?

Reply:
You're right it absolutely was in my mind that it does do this.
I will add a comment.